go/ethclient/geth_rpc_client.go

```diff
@@ -152,21 +149,19 @@ func (e *gethRPCClient) SubmitTransaction(tx types.TxData) (*types.Transaction,
 		panic(err)
```

Comment:
"wallet.SignTransaction" doesn't belong here. It should be called before broadcasting.
Let's change this as part of this PR as well, because it's quite confusing

Reply:
Long due a cleanup, done 👍